modules/_labs/combobox/react/spec/Combobox.spec.tsx

```diff
@@ -0,0 +1,325 @@
+import * as React from 'react';
+import Combobox from '../lib/Combobox';
+import {MenuItem} from '../../../menu/react/index';
+import {TextInput} from '../../../../text-input/react';
+import {render, fireEvent} from '@testing-library/react';
+
+describe('Combobox', () => {
+  const cb = jest.fn().mockImplementation((event: Event) => event);
+
+  afterEach(() => {
+    cb.mockReset();
+  });
+
+  test('Autocomplete items should show when focused', async () => {
+    const menuText = 'test 1';
+    const placeholderText = 'placeholder';
+    const autocompleteItems = [<MenuItem>{menuText}</MenuItem>, <MenuItem>{menuText}</MenuItem>];
+    const {queryByRole, findAllByText, findByPlaceholderText} = render(
+      <Combobox autocompleteItems={autocompleteItems}>
+        <TextInput placeholder={placeholderText} />
+      </Combobox>
+    );
+
+    expect(await queryByRole('listbox')).toBeNull();
+
+    fireEvent.focus(await findByPlaceholderText(placeholderText));
+    const menuItems = await findAllByText(menuText);
+
+    expect(await queryByRole('listbox')).toBeVisible();
+    expect(menuItems.length).toBe(2);
+  });
+
+  test('Selected autocomplete items should loop around using down arrow', async () => {
+    const id = 'my-id';
+    const placeholderText = 'placeholder';
+    const autocompleteItems = [<MenuItem />, <MenuItem />];
+    const {findByPlaceholderText} = render(
+      <Combobox id={id} autocompleteItems={autocompleteItems}>
+        <TextInput placeholder={placeholderText} />
+      </Combobox>
+    );
+
+    const input = await findByPlaceholderText(placeholderText);
+    const down = {keyCode: 40, key: 'ArrowDown'};
+
+    fireEvent.keyDown(input, down);
+    expect(input.getAttribute('aria-activedescendant')).toEqual(`${id}-option-0`);
+    fireEvent.keyDown(input, down);
+    expect(input.getAttribute('aria-activedescendant')).toEqual(`${id}-option-1`);
+    fireEvent.keyDown(input, down);
+    expect(input.getAttribute('aria-activedescendant')).toEqual(`${id}-option-0`);
+  });
```

Comment:
We'll probably migrate a lot of these tests to Cypress. It's fine as a unit test for now